### PR TITLE
bacnet-stack: init at 1.0.0

### PR DESCRIPTION
**Pull request by WhittlesJr**
###### Motivation for this change

I use BACnet MS/TP for my employment, and this is the only tool that I'm aware of that allows you to capture BACnet MS/TP traffic running on a serial port on Linux. It works great with Wireshark.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

I use this for the "mstpcap" executable which, I have confirmed, works. There are a number of tools that I don't use, and I have not tested them.

### CODE REVIEW

File: pkgs/tools/networking/bacnet-stack/default.nix
Comment (dali99) [inline pkgs/tools/networking/bacnet-stack/default.nix:11]: use `fetchFromGitHub`

File: pkgs/tools/networking/bacnet-stack/default.nix
Comment (dali99) [inline pkgs/tools/networking/bacnet-stack/default.nix:28]: All packages need a maintainer

File: pkgs/tools/networking/bacnet-stack/default.nix
Comment (dali99) [inline pkgs/tools/networking/bacnet-stack/default.nix:5]: When not using release numbers, packages should be versioned with unstable-2019-04-14

```suggestion
  name = "bacnet-stack";
  version = "unstable-2019-04-14";
```

File: pkgs/tools/networking/bacnet-stack/default.nix
Comment (dali99) [inline pkgs/tools/networking/bacnet-stack/default.nix:8]: Why from stargieg/bacnet-stack?

Seems to me the newest version is https://github.com/bacnet-stack/bacnet-stack

File: pkgs/tools/networking/bacnet-stack/default.nix
Comment (dali99) [inline pkgs/tools/networking/bacnet-stack/default.nix:22]:
```suggestion
  installPhase = ''
    mkdir $out
    cp -r bin $out/bin
  '';
```

**Review by D4ndellion — CHANGES_REQUESTED** [review]
Needs bumping to the newest version from the right repo source.

File: pkgs/tools/networking/bacnet-stack/default.nix
Comment (WhittlesJr) [reply pkgs/tools/networking/bacnet-stack/default.nix:8]: Good catch! I'm not sure how I googled my way to that other repo.

**Comment by WhittlesJr** [comment]
@dali99, thanks for the review. I believe I've applied the needed changes.

**Comment by dali99** [comment]
Commits need to be squshed, PR title needs updating.

`bacnet-stack: init at 1.0.0` should be the only commit

**Comment by WhittlesJr** [comment]
Should be good to go.

For future reference, should I always keep everything in a single commit during the review process? I figured that it would make reviewing easier if I maintained the history until the review was done. I am aware that PRs should be squashed to a single commit before merging, of course.

**Comment by dali99** [comment]
I'm also on the fence about it. I personally would like to avoid force pushes until the end, since it's much easier as you say to see the diffs, not to mention that force pushing has a tendency to break the links in the notifications.

But that's not what I usually see people doing.

**Review by D4ndellion — APPROVED** [review]
builds, haven't tested binaries

**Comment by WhittlesJr** [comment]
Anything remaining before we can merge?

**Comment by dali99** [comment]
no we just need someone with mergeaccess to look at it
@jonringer

File: pkgs/top-level/all-packages.nix
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:147]: unnecessary
```suggestion
  bacnet-stack = callPackage ../tools/networking/bacnet-stack {};
```

File: pkgs/tools/networking/bacnet-stack/default.nix
Comment (jonringer) [inline pkgs/tools/networking/bacnet-stack/default.nix:18]: this looks like a good candidate for `makeFlags`, but i couldn't get it to build after

Comment (WhittlesJr) [reply pkgs/tools/networking/bacnet-stack/default.nix:18]: Do you want me to figure that out before merging? Or can I leave it as-is?

Comment (jonringer) [reply pkgs/tools/networking/bacnet-stack/default.nix:18]: I couldn't figure it out... so either way, it's good.

**Comment by jonringer** [comment]
Please follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) and [manual#submitting-changes-making-patches ](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) and squash the fix-up commits.

This can be done without `git rebase -i` by doing:
```
git reset HEAD~1                    # move fix-up commits into unstaged
git add -- pkgs/                    # move changes into staged
git commit --amend --no-edit        # add changes to previous commit
git push ... ... --force            # modify current PR branch
```

However, `git rebase -i` is a more powerful alternative, I created a small video demonstrating it's use [here](https://www.youtube.com/watch?v=OR8Q-4PLalc). A more indepth text tutorial can be found [here](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase)

**Comment by WhittlesJr** [comment]
@jonringer, For future reference, should I always keep everything in a single commit during the review process? I figured that it would make reviewing easier if I maintained the history until the review was done. I am aware that PRs should be squashed to a single commit before merging, of course.

**Comment by WhittlesJr** [comment]
@jonringer Squashed and rebased onto master

**Comment by jonringer** [comment]
commits should be "logical units", and most people interpret that as, "if i were to cherry-pick these commits, what level abstraction would i expect"

For package additions, it's usually one per package

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 96872 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>bacnet-stack</li>  </ul></details>

**Pull request merged**
